app/actions/utils.ts

```diff
@@ -32,3 +33,90 @@ export const getStatusMessage = (status: AppNotificationStatus) => {
   }
   return null;
 };
+export const getKns = async (params: {
+  address: string;
+  contract: Contract;
+}): Promise<Domain> => {
+  const domain: any = {};
+
+  try {
+    const domainName = await params.contract.defaultNames(params.address);
+    // what do we do if this is false?
+    const isNameVerified =
+      (await params.contract.getDomainHolder(domainName)) === params.address;
+    if (!isNameVerified) null;
+    domain.name = `${domainName}.klima`;
+    const customImage = await params.contract.getDomainData(domainName);
+    if (customImage && JSON.parse(customImage).imgAddress) {
+      domain.image = JSON.parse(customImage).imgAddress;
```

Comment:
```suggestion
    if (customImage) {
      const data = JSON.parse(customImage);
      domain.image = data?.imgAddress;
```

This is probably not a costly parse but just for good practice, parse once and fallback to undefined or something?

Reply:
updated. I left the check for imageUrl because this call can return a twitter url or something and the user can also change they imgAddress key.

Reply:
updated